api/src/controllers/website.js

```diff
@@ -10,25 +10,21 @@ const { fillErrorObject } = require('../middleware/error');
  * Handle GET request from /clientWebsite/:team_id
  * @param {*} req request object, containing teamId
  * @param {*} res response object
- * @returns 200: the team's website info was found
- * @returns 404: team's website info is not found
+ * @returns 200: return the team's website info if found or return emptyWebsiteInfo if no found
  * @returns 500: server error
  */
-function getWebPageDetails(req, res) {
-  Website.findOne({
-    teamId: req.params.team_id,
-  })
-    .then((websiteInfo) => {
-      if (!websiteInfo) {
-        return res
-          .status(404)
-          .send(
-            'No Website info found for this Team (This is likely because they have not added any website information yet)',
-          );
-      }
-      return res.status(200).send(websiteInfo);
-    })
-    .catch((error) => res.status(500).json(`Server Error: ${error.message}`));
+async function getWebPageDetails(req, res, next) {
+  const { team_id } = req.params;
+  try {
+    const foundWebsiteInfo = await Website.findOne({ teamId: team_id });
+    if (foundWebsiteInfo) {
+      return res.status(200).json(foundWebsiteInfo);
+    }
+    const emptyWebsiteInfo = { teamId: team_id, pages: [] };
```

Comment:
I think a more scalable solution is to have an invariant where an empty list of pages is associated with a `Team` by default, so we don't have to interfere and force returning an empty page list to fix this issue. What do you reckon?
What about when a user first signs up, we initialize their webpages to empty for them? Would this be a cleaner approach?

Reply:
This basically ensures all state associated with a user's page is default initialized to appropriate values

Reply:
To make this even more scalable, we could consider implementing db triggers: https://www.google.com/search?q=db+trigger+mongodb&oq=db+trigger+mongodb&aqs=chrome..69i57j0i22i30j0i10i22i30j0i22i30l3.3014j0j7&client=ubuntu&sourceid=chrome&ie=UTF-8

so that the appropriate models get auto updated when a user signs up